internal/inspect/policy.go

```diff
@@ -65,18 +67,38 @@ func (pol *Policy) Inspect(p *policyv1.Policy) error {
 		return nil
 	}
 
-	referencedVariables, err := pol.inspectConditions(p)
+	referencedDerivedRoles, referencedVariables, err := pol.inspectDefinitionsAndRules(p)
 	if err != nil {
-		return fmt.Errorf("failed to inspect conditions of the policy %s: %w", policyID, err)
+		return fmt.Errorf("failed to inspect definitions and rules of the policy %s: %w", policyID, err)
 	}
+
+	var derivedRoles []*responsev1.InspectPoliciesResponse_DerivedRole
+	_, isDerivedRolePolicy := p.PolicyType.(*policyv1.Policy_DerivedRoles)
+	if isDerivedRolePolicy {
+		derivedRoles = policy.ListDerivedRoles(p)
```

Comment:
Rather than casting the policy twice (here and again in `ListDerivedRoles`) you can do something like:

```go
if drp := p.PolicyType.GetDerivedRoles(); drp != nil {
  ...
  derivedRoles = policy.ListDerivedRoles(policyID, drp)
  ...
} else {
  ...
}
```